### PR TITLE
GA VMLiveUpdateFeatures feature-gate

### PR DESCRIPTION
**Pull request by acardace**
### What this PR does
The `VMLiveUpdateFeatures` feature-gate has been introduced in KubeVirt v1.0.0 after being here for multiple releases and having undergone major cleanups and bug fixing I think is now ready to be GAed.

Note that this PR just GAs the feature-gate and doesn't change the default `vmRolloutStrategy` which remains `Stage`. 

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
GA the `VMLiveUpdateFeatures` feature-gate.
```

### CODE REVIEW

File: tests/hotplug/affinity.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ import (
 	"kubevirt.io/kubevirt/tests/testsuite"
 )
 
-var _ = Describe("[sig-compute]VM Affinity", decorators.SigCompute, decorators.SigComputeMigrations, decorators.RequiresTwoSchedulableNodes, decorators.VMLiveUpdateFeaturesGate, Serial, func() {
```
Comment (xpivarc) [inline tests/hotplug/affinity.go:33]: While `VMLiveUpdateFeaturesGate` doesn't represent the feature anymore can we just rename the decorator?

File: pkg/virt-controller/watch/vm/vm_test.go [hunk 2]
```diff
@@ -6751,20 +6730,20 @@ var _ = Describe("VirtualMachine", func() {
...
-				if strat == &liveUpdate && len(fgs) > 0 {
+				if strat == &liveUpdate {
 					vmi, err = virtFakeClient.KubevirtV1().VirtualMachineInstances(vm.Namespace).Get(context.TODO(), vm.Name, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
 					Expect(vmi.Spec.Domain.CPU.Sockets).To(Equal(uint32(4)))
 				}
 			},
 				Entry("should appear if the feature gate is not set",
-					[]string{}, &liveUpdate, restartRequiredMatcher(k8sv1.ConditionTrue)),
+					&liveUpdate, restartRequiredMatcher(k8sv1.ConditionTrue)),
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/vm/vm_test.go:6740]: This `Entry` is wrong and should be dropped entirely (FG is always enabled)

File: pkg/virt-controller/watch/vm/vm_test.go [hunk 3]
```diff
@@ -6751,20 +6730,20 @@ var _ = Describe("VirtualMachine", func() {
...
 				Entry("should appear if the VM rollout strategy is not set",
-					[]string{virtconfig.VMLiveUpdateFeaturesGate}, nil, restartRequiredMatcher(k8sv1.ConditionTrue)),
+					nil, restartRequiredMatcher(k8sv1.ConditionTrue)),
 				Entry("should appear if the VM rollout strategy is set to Stage",
-					[]string{virtconfig.VMLiveUpdateFeaturesGate}, &stage, restartRequiredMatcher(k8sv1.ConditionTrue)),
+					&stage, restartRequiredMatcher(k8sv1.ConditionTrue)),
 				Entry("should not appear if both the VM rollout strategy and feature gate are set",
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/vm/vm_test.go:6745]: This `Entry` description should be renamed:
`"should not appear if the VM rollout strategy is set to LiveUpdate"`

**Review by fossedihelm — COMMENTED** [review]
Leftover in unit test

File: tests/hotplug/affinity.go (hunk 1 above)
Comment (acardace) [reply tests/hotplug/affinity.go:33]: What for though? The tests would always be run now.

**Comment by fossedihelm** [comment]
@acardace Failures are relevant

File: tests/hotplug/affinity.go (hunk 1 above)
Comment (acardace) [reply tests/hotplug/affinity.go:33]: Ok, makes sense, I'll re-add it

**Review by xpivarc — COMMENTED** [review]
/approve

**Comment by vladikr** [comment]
Thanks @xpivarc .
/unhold

I'm not against this change. I just wanted to make sure that we are following https://github.com/kubevirt/community/pull/251 going forward.

**Pull request merged**